lib/Dialect/ESI/runtime/cpp/include/esi/Ports.h

```diff
@@ -35,7 +35,12 @@ class ChannelPort {
   ChannelPort(const Type *type) : type(type) {}
   virtual ~ChannelPort() { disconnect(); }
 
-  virtual void connect() { connectImpl(); }
+  /// Set up a connection to the accelerator. The buffer size is optional and
+  /// should be considered merely a hint. Individual implementations use it
+  /// however they like. The unit is number of messages of the port type.
+  virtual void connect(std::optional<size_t> bufferSize = std::nullopt) {
```

Comment:
Suggesting changing from size_t to unsigned; size_t is usually used to indicate the size of something in memory - which, as the comment indicates, this isn't.